packages/gatsby-plugin-image/src/node-apis/preprocess-source.ts

```diff
@@ -0,0 +1,46 @@
+import { PreprocessSourceArgs } from "gatsby"
+import { babelParseToAst } from "./parser"
+import path from "path"
+import { extractStaticImageProps } from "./parser"
+import { writeImages } from "./image-processing"
+const extensions: Array<string> = [`.js`, `.jsx`, `.tsx`]
+
+export async function preprocessSource({
+  filename,
+  contents,
+  pathPrefix,
+  cache,
+  reporter,
+  store,
+  createNodeId,
+  actions: { createNode },
+}: PreprocessSourceArgs): Promise<string> {
+  if (
+    !contents.includes(`StaticImage`) ||
+    !contents.includes(`gatsby-plugin-image`) ||
+    !extensions.includes(path.extname(filename))
+  ) {
+    return contents
+  }
+  const root = store.getState().program.directory
+
+  const cacheDir = path.join(root, `.cache`, `caches`, `gatsby-plugin-image`)
```

Comment:
We use this line 2 times, it's probably good to make this a util as a typo is easily made